File: Level 2/연속된 부분 수열의 합/solution.js
function solution(sequence, k) {
  const n = sequence.length;
  let [left, right] = [0, 0];
  let answer = [0, n];
  let sum = sequence[0];

  while (right !== n) {
    if (sum < k) {
      right++;
      sum += sequence[right];
    } else {
      if (sum === k && right - left < answer[1] - answer[0]) {
        answer = [left, right];
      }
      sum -= sequence[left];
      left++;
    }
  }

  return answer;
}

File: Level 2/연속된 부분 수열의 합/solution.py
def solution(sequence, k):
    n = len(sequence)
    answer = (0, n)
    left, right = 0, 0
    sum = sequence[0]

    while True:
        if sum < k:
            right += 1
            if right == n:
                break
            sum += sequence[right]
        else:
            if sum == k and right - left < answer[1] - answer[0]:
                answer = (left, right)
            sum -= sequence[left]
            left += 1

    return answer
